src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -339,12 +343,27 @@ public String getShortDescription() {
 
     @Exported
     @Override public boolean isConcurrentBuild() {
-        return !Boolean.FALSE.equals(concurrentBuild);
+        DisableConcurrentBuildsJobProperty p = getProperty(DisableConcurrentBuildsJobProperty.class);
+        if (p != null) {
+            return false;
+        } else {
+            /* settings compatibility */
+            return !Boolean.FALSE.equals(concurrentBuild);
```

Comment:
Can just become

```java
return true;
```

now, since `onLoad` would already have cleared the field.